### PR TITLE
Add support for channel restrictions

### PR DESCRIPTION
**Pull request by ThoreKr**
Directly related to #1192

### CODE REVIEW

File: defaults/config.js [hunk 1]
```diff
@@ -370,6 +370,35 @@ module.exports = {
...
+		// @default ""
+		//
+		pattern: "",
+
+		// Message to return to the user when trying to join a channel not matching the pattern, supports regular expressions
+		//
+		// @type  string
+		// @default "This client ist restricted to certain channels."
+		//
+		restrictMessage: "This client ist restricted to certain channels."
```
Comment (MaxLeiter) [inline defaults/config.js]: is* restricted

File: defaults/config.js [hunk 2]
```diff
@@ -370,6 +370,35 @@ module.exports = {
 		primaryKey: "uid"
 	},
 
+	// Restrict to only certain channels
+	//
+  // @type     object
```
Comment (MaxLeiter) [inline defaults/config.js]: indentation

**Comment by ThoreKr** [comment]
No idea what atom did with the indentation again, but it should be fixed now

**Comment by xPaw** [comment]
Basically what you did is simply not create a UI window for channel, but on the server, you're in the channel and it will send you messages and other stuff from that channel.

**Comment by ThoreKr** [comment]
Well, that's an issue indeed.
It's not possible to control the server and changing it isn't possible due to legacy reasons.
The only way to prevent users from joining certain channels would be in the client.

**Comment by astorije** [comment]
> I also have a hard time grasping why should it be client's job to filter channels.

Client? Isn't this PR trying to do it on the server (though I totally get the first part of your comment)? Do you mean it should be the job of `irc-framework`?

> It's not possible to control the server and changing it isn't possible due to legacy reasons.

What do you mean by that?

**Comment by ThoreKr** [comment]
Oh, you were reffering to the lounge server-side?
My bad, messed that up with considering lounge just as a client and the server as the actual IRC-Server.
Ideally it would limit the channel access for users in that part of the code, which would not be executed in the browser, but in the lounge-server-side.

**Comment by xPaw** [comment]
> Client? Isn't this PR trying to do it on the server 

Client as in an IRC *client*, and server as in an IRC server.

**Comment by YaManicKill** [comment]
I'm not sure how I feel about this...

I definitely get the issue you are trying to get around, but I'm not sure the lounge is the best place for this restriction, to be honest.

I guess we do already have a way to lock the network that the client can connect to, but ... I'm not sure.

**Comment by xPaw** [comment]
I'll close this as we all agree that doing this in the client is not the right place. Channel restrictions should be done on the IRC server.

**Pull request closed without merging**